src/utils.ts

```diff
@@ -1,41 +1,30 @@
-import config from 'config';
 import type { HttpClient } from './api/common';
-import type { EnvelopingConfig } from './common/config.types';
-import type { RelayInfo } from './common/relayHub.types';
-import { ENVELOPING_ROOT } from './constants/configs';
-import { BigNumberish, BigNumber, getDefaultProvider, Transaction } from 'ethers';
+import { BigNumberish, BigNumber, Transaction, constants } from 'ethers';
 import { BigNumber as BigNumberJs } from 'bignumber.js';
-import { RelayHub__factory } from '@rsksmart/rif-relay-contracts';
+import { ICustomSmartWalletFactory__factory, IERC20__factory, ISmartWalletFactory__factory, RelayHub__factory } from '@rsksmart/rif-relay-contracts';
 import log from 'loglevel';
-import type { DeployRequest, RelayRequest } from './common/relayRequest.types';
-import { isDeployTransaction } from './common/relayRequest.utils';
-import type { EnvelopingTxRequest } from './common/relayTransaction.types';
-import type { RequestConfig } from './RelayClient';
+import { EstimateInternalGasParams, getEnvelopingConfig, getProvider, isDeployRequest, isDeployTransaction, TokenGasEstimationParams } from './common';
+import type { RequestConfig, EnvelopingTxRequest, DeployRequest, RelayRequest, RelayInfo } from './common';
+import { MISSING_SMART_WALLET_ADDRESS, MISSING_CALL_FORWARDER } from './constants/errorMessages';
 
 
 const INTERNAL_TRANSACTION_ESTIMATED_CORRECTION = 20000; // When estimating the gas an internal call is going to spend, we need to substract some gas inherent to send the parameters to the blockchain
 const ESTIMATED_GAS_CORRECTION_FACTOR = 1;
 
-const getEnvelopingConfig = () => {
-  try {
-    return config.get<EnvelopingConfig>(ENVELOPING_ROOT);
-  } catch {
-    throw new Error(
-      `Could not read enveloping configuration. Make sure your configuration is nested under ${ENVELOPING_ROOT} key.`
-    );
-  }
-};
 
 const selectNextRelay = async (
   httpClient: HttpClient
 ): Promise<RelayInfo | undefined> => {
   const { preferredRelays } = getEnvelopingConfig();
 
-  for (const managerData of preferredRelays ?? []) {
+  for (const preferredRelay of preferredRelays ?? []) {
     let hubInfo;
+    let managerData;
 
     try {
-      hubInfo = await httpClient.getChainInfo(managerData.url);
+      hubInfo = await httpClient.getChainInfo(preferredRelay);
+      const relayHub = RelayHub__factory.connect(hubInfo.relayHubAddress, getProvider());
```

Comment:
Why don't we check any more for the `ready` property in `hubInfo`

Reply:
we do in line 33, but I did some refactor.

Reply:
updated